prometheus/promhttp/http.go

```diff
@@ -406,3 +418,49 @@ func httpError(rsp http.ResponseWriter, err error) {
 		http.StatusInternalServerError,
 	)
 }
+
+func GetWriter(r *http.Request, rsp http.ResponseWriter, disableCompression bool, offeredCompressions []Compression) (io.Writer, error) {
+	w := io.Writer(rsp)
+	rsp.Header().Set(contentEncodingHeader, "identity")
+	if !disableCompression {
+		offers := defaultCompressionFormats
+		if len(offeredCompressions) > 0 {
+			offers = offeredCompressions
```

Comment:
This should likely be done not in handler but when we create it. Somewhere here:

![image](https://github.com/prometheus/client_golang/assets/6950331/9529c1b2-46ea-467e-a688-1d4fffed92d3)

(unnecessary "handler" work that could be precomputed)

Reply:
Good point, I moved it to the handler